Compilation error with `mvn clean install`

Hi, I can't build the project by instructions list in README.
I got the following errors...

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project vertx-hawkular-metrics: Compilation failure: Compilation failure:
[ERROR] /Users/akuo/Developer/hawkular/vertx-hawkular-metrics/src/main/java/io/vertx/ext/hawkular/impl/HttpServerMetricsImpl.java:[31,8] io.vertx.ext.hawkular.impl.HttpServerMetricsImpl is not abstract and does not override abstract method responsePushed(java.lang.Void,io.vertx.core.http.HttpMethod,java.lang.String,io.vertx.core.http.HttpServerResponse) in io.vertx.core.spi.metrics.HttpServerMetrics
[ERROR] /Users/akuo/Developer/hawkular/vertx-hawkular-metrics/src/main/java/io/vertx/ext/hawkular/impl/HttpClientMetricsImpl.java:[37,8] io.vertx.ext.hawkular.impl.HttpClientMetricsImpl is not abstract and does not override abstract method requestReset(io.vertx.ext.hawkular.impl.HttpClientRequestMetrics) in io.vertx.core.spi.metrics.HttpClientMetrics
[ERROR] /Users/akuo/Developer/hawkular/vertx-hawkular-metrics/src/main/java/io/vertx/ext/hawkular/impl/DatagramSocketMetricsImpl.java:[34,8] io.vertx.ext.hawkular.impl.DatagramSocketMetricsImpl is not abstract and does not override abstract method listening(java.lang.String,io.vertx.core.net.SocketAddress) in io.vertx.core.spi.metrics.DatagramSocketMetrics
[ERROR] /Users/akuo/Developer/hawkular/vertx-hawkular-metrics/src/main/java/io/vertx/ext/hawkular/impl/DatagramSocketMetricsImpl.java:[47,3] method does not override or implement a method from a supertype
```
Did I do it wrong or there's a problem?

--- a/src/main/asciidoc/dataobjects.adoc
+++ b/src/main/asciidoc/dataobjects.adoc
@@ -1,5 +1,49 @@
 = Cheatsheets
 
+[[HawkularServerOptions]]
+== HawkularServerOptions
+
+++++
+ Options specific to a Hawkular server.
+++++
+'''
+
+[cols=">25%,^25%,50%"]
+[frame="topbot"]
+|===
+^|Name | Type ^| Description
+|[[id]]`id`|`String`|
++++
+Set the identifier used for authentication.
++++
+|[[persona]]`persona`|`String`|
++++
+Set the Hawkular Persona identifier. This is used to impersonate an organization with user credentials.
++++
+|[[secret]]`secret`|`String`|
++++
+Set the secret used for authentication.
++++
+|===
+
+[[StandaloneMetricsOptions]]
+== StandaloneMetricsOptions
+
+++++
+ Options specific to a standalone Metrics server.
+++++
+'''
+
+[cols=">25%,^25%,50%"]
+[frame="topbot"]
+|===
+^|Name | Type ^| Description
+|[[tenant]]`tenant`|`String`|
++++
+Set the Hawkular tenant. Defaults to <code>default</code>.
++++
+|===
+
 [[VertxHawkularOptions]]
 == VertxHawkularOptions
 
@@ -77,50 +121,6 @@ Set the Hawkular server type (standalone Metrics, Hawkular, ... etc). Defaults t
 |[[standaloneMetricsOptions]]`standaloneMetricsOptions`|`link:dataobjects.html#StandaloneMetricsOptions[StandaloneMetricsOptions]`|
 +++
 Set the options specific to a standalone Metrics server.
-+++
-|===
-
-[[HawkularServerOptions]]
-== HawkularServerOptions
-
-++++
- Options specific to a Hawkular server.
-++++
-'''
-
-[cols=">25%,^25%,50%"]
-[frame="topbot"]
-|===
-^|Name | Type ^| Description
-|[[id]]`id`|`String`|
-+++
-Set the identifier used for authentication.
-+++
-|[[persona]]`persona`|`String`|
-+++
-Set the Hawkular Persona identifier. This is used to impersonate an organization with user credentials.
-+++
-|[[secret]]`secret`|`String`|
-+++
-Set the secret used for authentication.
-+++
-|===
-
-[[StandaloneMetricsOptions]]
-== StandaloneMetricsOptions
-
-++++
- Options specific to a standalone Metrics server.
-++++
-'''
-
-[cols=">25%,^25%,50%"]
-[frame="topbot"]
-|===
-^|Name | Type ^| Description
-|[[tenant]]`tenant`|`String`|
-+++
-Set the Hawkular tenant. Defaults to <code>default</code>.
 +++
 |===
 

--- a/src/main/java/io/vertx/ext/hawkular/impl/DatagramSocketMetricsImpl.java
+++ b/src/main/java/io/vertx/ext/hawkular/impl/DatagramSocketMetricsImpl.java
@@ -15,8 +15,7 @@
  */
 package io.vertx.ext.hawkular.impl;
 
-import io.vertx.core.net.SocketAddress;
-import io.vertx.core.spi.metrics.DatagramSocketMetrics;
+import static java.util.stream.Collectors.*;
 
 import java.util.Map;
 import java.util.Map.Entry;
@@ -24,7 +23,9 @@ import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
 import java.util.concurrent.atomic.LongAdder;
 
-import static java.util.stream.Collectors.*;
+import io.vertx.core.net.SocketAddress;
+import io.vertx.core.net.impl.SocketAddressImpl;
+import io.vertx.core.spi.metrics.DatagramSocketMetrics;
 
 /**
  * Implementation of {@link DatagramSocketMetrics} which relays data to {@link DatagramSocketMetricsSupplier}.
@@ -45,8 +46,8 @@ public class DatagramSocketMetricsImpl implements DatagramSocketMetrics {
   }
 
   @Override
-  public void listening(SocketAddress localAddress) {
-    this.localAddress = localAddress;
+  public void listening(String localName, SocketAddress localAddress) {
+    this.localAddress = new SocketAddressImpl(localAddress.port(), localName);
   }
 
   @Override

--- a/src/main/java/io/vertx/ext/hawkular/impl/EventBusMetricsImpl.java
+++ b/src/main/java/io/vertx/ext/hawkular/impl/EventBusMetricsImpl.java
@@ -15,21 +15,20 @@
  */
 package io.vertx.ext.hawkular.impl;
 
-import io.vertx.core.eventbus.ReplyFailure;
-import io.vertx.core.spi.metrics.EventBusMetrics;
-
 import java.util.ArrayList;
 import java.util.List;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
 import java.util.concurrent.atomic.LongAdder;
 
+import io.vertx.core.eventbus.ReplyFailure;
+import io.vertx.core.spi.metrics.EventBusMetrics;
+
 /**
  * @author Thomas Segismont
  */
 public class EventBusMetricsImpl implements EventBusMetrics<EventBusHandlerMetrics>, MetricSupplier {
   private final String baseName;
-  private final Scheduler scheduler;
   private final LongAdder handlers = new LongAdder();
   private final ConcurrentMap<String, HandlersMeasurements> handlersMeasurements = new ConcurrentHashMap<>(0);
   private final LongAdder errorCount = new LongAdder();
@@ -52,10 +51,8 @@ public class EventBusMetricsImpl implements EventBusMetrics<EventBusHandlerMetri
   private final LongAdder deliveredRemoteMessages = new LongAdder();
   private final LongAdder replyFailures = new LongAdder();
 
-  public EventBusMetricsImpl(String prefix, Scheduler scheduler) {
+  public EventBusMetricsImpl(String prefix) {
     baseName = prefix + (prefix.isEmpty() ? "" : ".") + "vertx.eventbus.";
-    this.scheduler = scheduler;
-    scheduler.register(this);
   }
 
   @Override
@@ -217,7 +214,6 @@ public class EventBusMetricsImpl implements EventBusMetrics<EventBusHandlerMetri
 
   @Override
   public void close() {
-    scheduler.unregister(this);
   }
 
 }

--- a/src/main/java/io/vertx/ext/hawkular/impl/HttpClientMetricsImpl.java
+++ b/src/main/java/io/vertx/ext/hawkular/impl/HttpClientMetricsImpl.java
@@ -16,6 +16,13 @@
 
 package io.vertx.ext.hawkular.impl;
 
+import static java.util.stream.Collectors.*;
+
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
 import io.vertx.core.http.HttpClientRequest;
 import io.vertx.core.http.HttpClientResponse;
 import io.vertx.core.http.WebSocket;
@@ -23,13 +30,6 @@ import io.vertx.core.net.SocketAddress;
 import io.vertx.core.net.impl.SocketAddressImpl;
 import io.vertx.core.spi.metrics.HttpClientMetrics;
 import io.vertx.ext.hawkular.impl.HttpClientConnectionsMeasurements.Snapshot;
-
-import java.util.Map;
-import java.util.Map.Entry;
-import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.ConcurrentMap;
-
-import static java.util.stream.Collectors.*;
 
 /**
  * @author Thomas Segismont
@@ -52,6 +52,15 @@ public class HttpClientMetricsImpl implements HttpClientMetrics<HttpClientReques
     HttpClientRequestMetrics httpClientRequestMetrics = new HttpClientRequestMetrics(key);
     httpClientRequestMetrics.resetTimer();
     return httpClientRequestMetrics;
+  }
+
+  @Override
+  public HttpClientRequestMetrics responsePushed(SocketAddress socketMetric, SocketAddress localAddress, SocketAddress remoteAddress, HttpClientRequest request) {
+    return null;
+  }
+
+  @Override
+  public void requestReset(HttpClientRequestMetrics requestMetric) {
   }
 
   @Override

--- a/src/main/java/io/vertx/ext/hawkular/impl/HttpServerMetricsImpl.java
+++ b/src/main/java/io/vertx/ext/hawkular/impl/HttpServerMetricsImpl.java
@@ -15,15 +15,16 @@
  */
 package io.vertx.ext.hawkular.impl;
 
+import static java.util.concurrent.TimeUnit.*;
+
+import java.util.concurrent.atomic.LongAdder;
+
+import io.vertx.core.http.HttpMethod;
 import io.vertx.core.http.HttpServerRequest;
 import io.vertx.core.http.HttpServerResponse;
 import io.vertx.core.http.ServerWebSocket;
 import io.vertx.core.net.SocketAddress;
 import io.vertx.core.spi.metrics.HttpServerMetrics;
-
-import java.util.concurrent.atomic.LongAdder;
-
-import static java.util.concurrent.TimeUnit.*;
 
 /**
  * @author Thomas Segismont
@@ -56,6 +57,15 @@ public class HttpServerMetricsImpl implements HttpServerMetrics<Long, Void, Void
   public Long requestBegin(Void socketMetric, HttpServerRequest request) {
     requests.increment();
     return System.nanoTime();
+  }
+
+  @Override
+  public void requestReset(Long requestMetric) {
+  }
+
+  @Override
+  public Long responsePushed(Void socketMetric, HttpMethod method, String uri, HttpServerResponse response) {
+    return null;
   }
 
   @Override

--- a/src/main/java/io/vertx/ext/hawkular/impl/VertxMetricsImpl.java
+++ b/src/main/java/io/vertx/ext/hawkular/impl/VertxMetricsImpl.java
@@ -16,6 +16,8 @@
 
 package io.vertx.ext.hawkular.impl;
 
+import java.util.Collections;
+
 import io.vertx.core.Context;
 import io.vertx.core.Vertx;
 import io.vertx.core.datagram.DatagramSocket;
@@ -39,50 +41,42 @@ import io.vertx.core.spi.metrics.HttpServerMetrics;
 import io.vertx.core.spi.metrics.TCPMetrics;
 import io.vertx.ext.hawkular.VertxHawkularOptions;
 
-import java.util.Collections;
-
 /**
  * Metrics SPI implementation.
  *
  * @author Thomas Segismont
  */
 public class VertxMetricsImpl extends DummyVertxMetrics {
-  private final String prefix;
+  private final Vertx vertx;
+  private final VertxHawkularOptions options;
   private final HttpServerMetricsSupplier httpServerMetricsSupplier;
   private final HttpClientMetricsSupplier httpClientMetricsSupplier;
   private final NetServerMetricsSupplier netServerMetricsSupplier;
   private final NetClientMetricsSupplier netClientMetricsSupplier;
   private final DatagramSocketMetricsSupplier datagramSocketMetricsSupplier;
-  private final VertxHawkularOptions options;
+  private final EventBusMetricsImpl eventBusMetrics;
 
-  private final Sender sender;
-  private final Scheduler scheduler;
+  private Sender sender;
+  private Scheduler scheduler;
 
   /**
    * @param vertx   the {@link Vertx} managed instance
    * @param options Vertx Hawkular options
    */
   public VertxMetricsImpl(Vertx vertx, VertxHawkularOptions options) {
-    prefix = options.getPrefix();
+    this.vertx = vertx;
     this.options = options;
+    String prefix = options.getPrefix();
     httpServerMetricsSupplier = new HttpServerMetricsSupplier(prefix);
     httpClientMetricsSupplier = new HttpClientMetricsSupplier(prefix);
     netServerMetricsSupplier = new NetServerMetricsSupplier(prefix);
     netClientMetricsSupplier = new NetClientMetricsSupplier(prefix);
     datagramSocketMetricsSupplier = new DatagramSocketMetricsSupplier(prefix);
-    Context context = vertx.getOrCreateContext();
-    sender = new Sender(vertx, options, context);
-    scheduler = new Scheduler(vertx, options, context, sender);
-    scheduler.register(httpServerMetricsSupplier);
-    scheduler.register(httpClientMetricsSupplier);
-    scheduler.register(netServerMetricsSupplier);
-    scheduler.register(netClientMetricsSupplier);
-    scheduler.register(datagramSocketMetricsSupplier);
+    eventBusMetrics = new EventBusMetricsImpl(prefix);
   }
 
   @Override
-  public HttpServerMetrics<Long, Void, Void> createMetrics(HttpServer server, SocketAddress localAddress,
-                                                           HttpServerOptions options) {
+  public HttpServerMetrics<Long, Void, Void> createMetrics(HttpServer server, SocketAddress localAddress, HttpServerOptions options) {
     return new HttpServerMetricsImpl(localAddress, httpServerMetricsSupplier);
   }
 
@@ -108,7 +102,7 @@ public class VertxMetricsImpl extends DummyVertxMetrics {
 
   @Override
   public EventBusMetrics createMetrics(EventBus eventBus) {
-    return new EventBusMetricsImpl(prefix, scheduler);
+    return eventBusMetrics;
   }
 
   @Override
@@ -122,18 +116,18 @@ public class VertxMetricsImpl extends DummyVertxMetrics {
   }
 
   @Override
-  public void close() {
-    scheduler.unregister(httpServerMetricsSupplier);
-    scheduler.unregister(httpClientMetricsSupplier);
-    scheduler.unregister(netServerMetricsSupplier);
-    scheduler.unregister(netClientMetricsSupplier);
-    scheduler.unregister(datagramSocketMetricsSupplier);
-    scheduler.stop();
-    sender.stop();
-  }
-
-  @Override
   public void eventBusInitialized(EventBus bus) {
+    // Finish setup
+    Context context = vertx.getOrCreateContext();
+    sender = new Sender(vertx, options, context);
+    scheduler = new Scheduler(vertx, options, context, sender);
+    scheduler.register(httpServerMetricsSupplier);
+    scheduler.register(httpClientMetricsSupplier);
+    scheduler.register(netServerMetricsSupplier);
+    scheduler.register(netClientMetricsSupplier);
+    scheduler.register(datagramSocketMetricsSupplier);
+    scheduler.register(eventBusMetrics);
+
     //Configure the metrics bridge. It just transforms the received metrics (json) to a Single Metric to enqueue it.
     if (options.isMetricsBridgeEnabled() && options.getMetricsBridgeAddress() != null) {
       bus.consumer(options.getMetricsBridgeAddress(), message -> {
@@ -157,6 +151,18 @@ public class VertxMetricsImpl extends DummyVertxMetrics {
         sender.handle(Collections.singletonList(dataPoint));
       });
     }
+  }
+
+  @Override
+  public void close() {
+    scheduler.unregister(httpServerMetricsSupplier);
+    scheduler.unregister(httpClientMetricsSupplier);
+    scheduler.unregister(netServerMetricsSupplier);
+    scheduler.unregister(netClientMetricsSupplier);
+    scheduler.unregister(datagramSocketMetricsSupplier);
+    scheduler.unregister(eventBusMetrics);
+    scheduler.stop();
+    sender.stop();
   }
 
 

